data2rdf/models/mapping.py

```diff
@@ -0,0 +1,234 @@
+"""Mapping models for data2rdf"""
+import json
+import warnings
+from abc import abstractmethod
+from typing import Any, Dict, Optional, Union
+
+from pydantic import AnyUrl, BaseModel, Field, ValidationInfo, field_validator
+from rdflib import Graph
+
+from data2rdf.config import Config
+from data2rdf.qudt.utils import _get_query_match
+from data2rdf.utils import is_float, is_integer, make_prefix
+
+
+class BasicConceptMapping(BaseModel):
+    """Basic mapping for any entity"""
+
+    key: Optional[str] = Field(
+        None, description="Key of the concept in the file"
+    )
+    iri: AnyUrl = Field(
+        ..., description="Ontological class related to this concept"
+    )
+    config: Config = Field(
```

Comment:
Well, the mapping models in this module should be treated as standalone in order to be independently reused in different parsers, especially the `QuantityMappings`. 

The `Config` object here is inherited to all child classes of the `BasicConceptMapping` since all of them are depending on it for building the subgraph. In order to ensure that the same config is used everywhere in the overall package, the same config object is later passed from the pipeline to the parser and from the parser to the mapping.

Does this make sense to you?

Reply:
Have you considered these approaches?
1. Have an upper class that contains the config and the entries (i.e., the `ConceptMapping` instances) as fields.
2. Put the config instance in the `__init__.py` so that it accessible to all the instances.

I do think your approach is valid, but was just wondering about the alternatives.

Reply:
Overwriting the `__init__` for a pydantic `BaseModel` is rather something which should be avoided and only be done if really necessary, like we do e.g in the DSMS SDK. 

The `default_factory` in this model here is making an `Config`-instance for your model if not provided by the child classes/ by the user. This is generlly a better practise in pydantic.

This actually does the same job as instanciating at `__init__` if not provided as `kwarg` by the child classes.

Is this what you tried to explain as alternative?